core/src/test/java/org/testcontainers/utility/DirectoryTarResourceTest.java

```diff
@@ -61,6 +63,45 @@ public void simpleRecursiveFileWithPermissionTest() {
                 exactlyNItems(1, allOf(containsString("-rwxr-xr--"), containsString("foo"))));
     }
 
+    @Test
+    public void transferFileDockerDaemon() {
+        GenericContainer container = new GenericContainer(
+                new ImageFromDockerfile()
+                        .withDockerfileFromBuilder(builder ->
+                                builder.from("alpine:3.3")
+                                        .copy("bar", "/foo/")
+                                        .cmd("ls", "-al", "/foo")
+                                        .build()
+                        ).withFileFromFile("bar", new File("src/test/resources/mappable-resource/test-resource.txt"),
+                        0754))
+                .withStartupCheckStrategy(new OneShotStartupCheckStrategy());
+
+        container.start();
+        String listing = container.getLogs();
+
+        assertThat("Listing shows that file is copied.",
+                Arrays.asList(listing.split("\\n")),
+                exactlyNItems(1, allOf(containsString("-rwxr-xr--"), containsString("bar"))));
+    }
+
+    @Test
+    public void unableToTransferFileNamedDotToDockerDaemon() {
+        VisibleAssertions.assertThrows("Unable to store file '" +
+                Paths.get("src", "test", "resources", "mappable-resource", "test-resource.txt") + "' to docker path '.'",
+            IllegalArgumentException.class, () -> {
+                new GenericContainer(
```

Comment:
this test can use `ImageFromDockerfile()` + `resolve()`, no need to construct the container

Reply:
`.resolve()` is protected so I have moved this test into `ImageFromDockerfileTest`